### ISSUE TITLE
Redesign the options service

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#40]
### Brief bug description

The IOptionsService interface dissects between production and development. However, the use of IOptionsMonitor vs IOptionsSnapshot should be driven by the characteristics of the options consumer (singleton vs transient/scoped).

### Repro steps

2. Click on '...'
3. Scroll down to '...'

### Expected behavior

What is the correct behavior?

### Test environment

 - Platform/OS: [e.g. .NET Core 3.1, iOS, Windows]
 - Browser [e.g. Chrome, Safari, Edge Chromium, Edge]
 - Version [e.g. 81]

### Additional context

Add any other context about the problem here. If you found an issue where a similar bug has been reported, add a link here.

### Screenshots

Add links to screenshots, if you deem it appropriate.

### Issue expands upon

This issue expands upon another issue(s): #`issue number`

### Blockers

This issue is blocked by the following issue(s): #`issue number`
